packages/automerge-repo-network-websocket/src/NodeWSServerAdapter.ts

```diff
@@ -30,11 +30,16 @@ export class NodeWSServerAdapter
   connect(peerId: PeerId) {
     this.peerId = peerId
     this.server.on("connection", (socket) => {
-      // When a socket closes, or disconnects, remove it from the array.
-      // TODO
-      //      socket.on("close", () => {
-      //        this.sockets = this.sockets.filter((s) => s !== socket)
-      //})
+      // When a socket closes, or disconnects, remove it from our list
+      socket.on("close", () => {
+        for (const [otherPeerId, otherSocket] of Object.entries(this.sockets)) {
+          if(socket === otherSocket) {
+            this.emit("peer-disconnected", { peerId: otherPeerId as PeerId })
```

Comment:
my hope had been at one point that we might have a more principled way of managing the set of peers for a given document but i can't really see how to avoid copying the peers list into every docsynchronizer. this is reasonable.

Reply:
we want at least the equivalent of this code for tabs in the MessageChannelNetworkAdapter as well, but AFAICT there's no way to actually detect a MessageChannel disconnection :(